public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -146,16 +170,27 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ data, query, dimensio
               onChange={(v) => onFieldChange(index, 'operator', v.value ?? '')}
               allowCustomValue
             />
-            <Select
-              menuShouldPortal
-              placeholder="Select value"
-              value={filter.filter ? filter.filter : ''}
-              allowCustomValue
-              options={getValidFilterOptions(filter.filter, filter.dimension)}
-              onChange={(v) => onFilterInputChange(index, v)}
-              isClearable
-            />
-
+            {filter.operator === 'eq' || filter.operator === 'ne' ? (
+              <MultiSelect
+                menuShouldPortal
+                placeholder="Select value(s)"
+                value={filter.filters}
+                options={getValidMultiSelectOptions(filter.filters, filter.dimension)}
+                onChange={(v) => onMultiSelectFilterChange(index, v)}
+                aria-label={'dimension-labels-select'}
+                allowCustomValue
+              />
+            ) : (
+              <Select
```

Comment:
I am not sure why you can only set one filter when using  `starts with`? Is that an API limitation? (Maybe you can explain that in a comment)

Reply:
This is an API limitation, will add a comment to explain.